pkgs/development/libraries/libfprint-2-tod1-elan/default.nix

```
@@ -0,0 +1,36 @@
+{ stdenv, lib, fetchzip, fetchgit, libfprint-tod, openssl, unzip }:
+
+stdenv.mkDerivation {
+  pname = "libfprint-2-tod1-elan";
+  version = "0.0.8";
+
+
+  src = fetchzip {
+    url = "https://download.lenovo.com/pccbbs/mobiles/r1slf01w.zip";
+    sha256 = "sha256-eY3QsoiBMVuIlT9XY2wI5kc89opgHE7nQa9n71vFpsM=";
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  buildPhase = ''
+    unzip libfprint-2-tod1-elan_0.0.8_Ubuntu22.04.zip
+    patchelf \
+      --set-rpath ${lib.makeLibraryPath [ libfprint-tod openssl]} \
+      libfprint-2-tod1-elan.so
+  '';
+
+  installPhase = ''
+    mkdir -p "$out/lib/libfprint-2/tod-1/"
+    cp libfprint-2-tod1-elan.so "$out/lib/libfprint-2/tod-1/"
+  '';
+
+  passthru.driverPath = "/lib/libfprint-2/tod-1";
+
+  meta = with lib; {
+    description = "Elan(04f3:0c4b) driver module for libfprint-2-tod Touch OEM Driver";
+    homepage = "https://support.lenovo.com/us/en/downloads/ds560939-elan-fingerprint-driver-for-ubuntu-2204-thinkpad-e14-gen-4-e15-gen-4";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

According to the documentation, you'll have to add yourself as maintainer (or someone else if you find someone willing to maintain the package).

- https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#quick-start-to-adding-a-package (far down from this subheader)
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

You'll also have to add yourself in a separate commit to `lib/maintainers.nix`. You can see some good examples of how this should be done among the [merged first contribution PRs](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+is%3Amerged+label%3A%228.has%3A+package+%28new%29%22+label%3A%2212.+first-time+contribution%22+)

```suggestion
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
    maintainers = with maintainers; [ qdlmcfresh ];
```

---

Ok did that

---

Cool! If you could rearrange the commits so that the maintainer one comes before the package one, I think this comment can be marked as resolved.